src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -797,9 +797,9 @@ public boolean isValid( final Object verificationToken )
 		}
 
 		@Override
-		public void postChange( final IBaseMonitor<T> monitor, final Iterable<T> change, final BaseActionSource source )
+		public void postChange( final IBaseMonitor<T> monitor, final Iterable<T> change, final IActionSource source )
 		{
-			if( source == TileChest.this.mySrc || ( source instanceof PlayerSource && ( (PlayerSource) source ).via == TileChest.this ) )
+			if( source == TileChest.this.mySrc || ( source.machine().isPresent() && source.machine().get() == TileChest.this ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3655)